src/core/dfn-abbr.js

```diff
@@ -0,0 +1,63 @@
+// @ts-check
+// Module core/dfn
+// - Finds all <dfn> elements with data-abbr attribute and parses them and registers definitions.
+// - Also populates abbrMap used by core/inlines for such definitions.
+
+import { registerDefinition } from "./dfn-map";
+
+export const name = "core/dfn-abbr";
+export const abbrMap = new Map();
+
+/**
+ * Generates simple Abbreviations from text passed.
+ *
+ * @param {String} text A string like "Permanent Account Number".
+ * @returns {String} abbr Abbreviation like "PAN"
+ */
+
+function getAbbreviationFromText(text) {
+  return text
+    .match(/\b([a-z])/gi)
+    .join("")
+    .toUpperCase();
+}
+
+/**
+ * Parses Elements of form <dfn data-abbr="PAN">Permanent Account Number</dfn>
+ * to <dfn data-abbr="PAN">Permanent Account Number (PAN)</dfn>
+ * @param {HTMLElement} dfn Element to be parsed.
+ */
+function parseAbbreviatedDefinition(dfn) {
+  // checks if text content is already in the form Permanent Account Number (PAN)
+  const matched = /\((.*?)\)/.exec(dfn.textContent);
```

Comment:
I don’t think we need this. The author ops into this behavior by explicitly using data-abbr.

Reply:
The author may do something like 
```<dfn data-abbr>Permanent Account Number (PAN)</dfn>``` 
or something like ```<dfn data-abbr>Statement of Purpose (SoP)</dfn>```  our another where our logic could get it wrong. He would still want that to be ```<abbr>```

Reply:
Sure, but she could have just done:

> `<dfn>Statement of Purpose (<abbr title="Statement of Purpose">SoP</abbr>)</dfn>` 

As this is a new feature, we should refrain from dealing with a case that literally doesn’t exist.

Reply:
guess what. ```<dfn data-abbr="PoR">Position of Responsibility</dfn>``` works just as good. Correcting.

Reply:
Corrected. Should be good to go now.